rbc/external.py

```diff
@@ -0,0 +1,60 @@
+# This code has heavily inspired in the numba.extending.intrisic code
+
+from numba.core import extending, funcdesc, types
+from rbc.typesystem import Type
+
+
+class _External:
+    def __init__(self, symbol, retty, argtys):
+        self.symbol = symbol
+        self.signature = retty(*argtys)
+
+    def _type_infer(self):
+        from numba.core.typing.templates import (
+            make_concrete_template,
+            infer_global,
+            infer,
+        )
+
+        template = make_concrete_template(
+            self.symbol, key=self.symbol, signatures=[self.signature]
+        )
+        infer(template)
+        infer_global(self, types.Function(template))
+
+    def _lower_external_call(self):
+        def codegen(context, builder, sig, args):
+            fndesc = funcdesc.ExternalFunctionDescriptor(
+                self.symbol, sig.return_type, sig.args
+            )
+            func = context.declare_external_function(builder.module, fndesc)
+            return builder.call(func, args)
+
+        extending.lower_builtin(self.symbol, *self.signature.args)(codegen)
```

Comment:
Are these methods used by numba? If not, merge the code into `register` method because `_type_infer` and `_lower_external_call` would represent numba internal API that should be irrelevant for rbc.